rmgpy/molecule/resonanceTest.py

```diff
@@ -314,3 +314,92 @@ def testKekulizedResonanceIsomersFused(self):
                 else:  # didn't break
                     self.fail(("Expected a resonance form {0!r} that was not generated.\n"
                               "Only generated these:\n{1}").format(expected, '\n'.join([repr(g) for g in isomers])))
+
+
+class ClarTest(unittest.TestCase):
+    """
+    Contains unit tests for Clar structure methods.
+    """
+
+    def setUp(self):
+        pass
+
+    def testClarTransformation(self):
+        """
+        Basic test that aromatic ring is generated.
+        """
+        mol = Molecule().fromSMILES('c1ccccc1')
+        sssr = mol.getSmallestSetOfSmallestRings()
+        clarTransformation(mol, sssr[0])
+        mol.updateAtomTypes()
+
+        self.assertTrue(mol.isAromatic())
+
+    def testClarOptimization(self):
+        """Test to ensure pi electrons are conserved during optimization"""
+        mol = Molecule().fromSMILES('C1=CC=C2C=CC=CC2=C1')  # Naphthalene
+        output = clarOptimization(mol)
+
+        for o in output:
+            SSSR = o[1]
+            bonds = o[2]
+            solution = o[3]
+
+            # Count pi electrons in molecule
+            pi = 0
+            for bond in bonds:
+                if bond.order == 'D':
+                    pi += 2
+
+            # Count pi electrons in solution
+            y = solution[0:len(SSSR)]
+            x = solution[len(SSSR):]
+            pi_solution = 6 * sum(y) + 2 * sum(x)
+
+            self.assertEqual(pi, pi_solution)
+
+    def testPhenanthrene(self):
+        """
+        Test phenanthrene, which is a basic case that should work.
+        """
+        mol = Molecule().fromSMILES('C1=CC=C2C(C=CC3=CC=CC=C32)=C1')
+        newmol = generateClarStructures(mol)
+
+        self.assertEqual(len(newmol), 1)
+        self.assertTrue(newmol[0].isAromatic())
```

Comment:
the test here should be more detailed: e.g, which ring is aromatic, which is not, (only testing overall aromaticity is not enough)